x/oracle/keeper/oracle.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/x/oracle/types"
+)
+
+func (k Keeper) RegisterOracle(ctx sdk.Context, oracle types.Oracle) error {
+	oracleAddr, err := sdk.AccAddressFromBech32(oracle.Address)
+	if err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	oracleKey := types.GetKeyPrefixOracle(oracleAddr)
+	if store.Has(oracleKey) {
+		return sdkerrors.Wrapf(types.ErrOracleAlreadyExist, "the oracle %s is already exists", oracle.Address)
+	}
+
+	oraclePubKey, err := k.accountKeeper.GetPubKey(ctx, oracleAddr)
+	if err != nil {
+		return err
+	}
+	if oraclePubKey == nil {
+		return sdkerrors.ErrKeyNotFound
```

Comment:
Instead of returning `sdkerrors.ErrKeyNotFound`, let's return other error. Because it's an error when the key doesn't exist, not pub key. 

Apart from that, is it required that the pub key of oracle exists? IMHO, it was required for the past because the seller should encrypt data with the oracle pub key. For now, however, we use TLS communication which is decrypted only in SGX so I think it is not required anymore.

Reply:
Ah, sorry please forget this comment. It is required to verify the signature in `DataCert`. Let's keep it this way :)